docs/integration/frameworks/set-up-with-nextjs.md

```diff
@@ -0,0 +1,95 @@
+---
+description: A step-by-step guide to integrate Keyshade into your Next.js app for managing secrets and environment variables — no .env files needed.
+---
+
+# How to Set Up Keyshade into a Next.js Project
+
+## 1. Create a Next.js Project
+
+Create a new Next.js app and make sure you select TypeScript:
+
+```bash
+npx create-next-app my-app
+```
+You can find more info about creating a Next.js project [here](https://nextjs.org/docs/app/api-reference/cli/create-next-app).
+
+## 2. Install the Keyshade CLI
+Install globally via npm:
+```bash
+npm i -g @keyshade/cli
+```
+_Refer: [Installing the CLI](/docs/getting-started/installing-the-cli)_
+
+## 3. Set Up Your Keyshade Profile
+If you're using Keyshade **for the first time**, refer: [Setting up your profile](/docs/getting-started/setting-up-your-profile)
+
+If you've **used Keyshade before**, you can switch to an existing profile by running:
+
+```bash
+keyshade profile use <profile_name>
+```
+
+Verify your profile is active:
+```bash
+keyshade workspace list
+```
+
+## 4. Create a Project in the Dashboard and Adding Secrets and Variables
+_Refer: [Creating a project](/docs/getting-started/adding-your-first-secret-and-variable)_
+
+> 💡 **Secrets vs Variables:**
+> 
+> -   **Secrets** are sensitive credentials like API keys or tokens.
+>     
+> -   **Variables** are non-sensitive configs like ports, flags, or feature toggles.
+>     
+
+Use `NEXT_PUBLIC_` prefix for any value exposed to the browser as this is a Next.js convention.
```

Comment:
hyper link the "Next.js convention" part to `https://nextjs.org/docs/app/guides/environment-variables#bundling-environment-variables-for-the-browser`